### PR TITLE
Upgrade to 4.8 upstream version and use latest helpers

### PR DESCRIPTION
**Pull request by maniackcrudelis**
## Problem
- Upgrade to wordpress 4.8
- Using the last helpers
- Following the new example app
- Fix issues https://github.com/YunoHost-Apps/wordpress_ynh/issues/24 and https://github.com/YunoHost-Apps/wordpress_ynh/issues/21

## Solution
- Change the upstream version
- Force language with wp-cli
- Force upgrade important plugins
- Force auto-upgrade with a simple plugin, and allow each user to manage their upgrades.

## PR Status
Work finished. Package_check, basic tests and upgrade from last version OK.
Could be reviewed and tested.

## Validation
---
- [ ] ~**Complete test** :~
- [X] ~**Upgrade previous version** : JimboJoe~
- [X] ~**Code review** : JimboJoe~
- [X] ~**Approval (LGTM)** : JimboJoe~
- [x] ~**Approval (LGTM)** : frju365~
- [X] ~**CI succeeded** : JimboJoe~

Since this PR has been rerouted from master to testing. It passes to a minor decision.
- [x] **Upgrade previous version** : JimboJoe
- [x] **Code review** : JimboJoe
- [x] **Approval (LGTM)** : JimboJoe
- [x] **Approval (LGTM)** : frju365
- [x] **CI succeeded** : JimboJoe
When the PR is mark as ready to merge, you have to wait for 3 days before really merge it.

### CODE REVIEW

**Review by JimboJoe — COMMENTED** [review]
Code review is fine for me!

Package_check failed the multisite site in "test with multisite series", which I guess is normal, so multi_instance should be set to 0 in that case, shouldn't it?
It also failed restoration in the "test with multisite series" (lxc container upgraded to latest yunohost version which should fix restoration bug), logs [here](https://paste.yunohost.org/urujimiqap.django).

I tested an upgrade from a freshly installed instance (with current version), which was successful, but wordpress was still in 4.7 version... ?!
(update to 4.8 from admin console went OK afterwards).

**Comment by maniackcrudelis** [comment]
> Package_check failed the multisite site in "test with multisite series", which I guess is normal, so multi_instance should be set to 0 in that case, shouldn't it?

You meant, the multi-instance test I guess. It doesn't normal, so I made a test.
This test works fine for me.

> It also failed restoration in the "test with multisite series" (lxc container upgraded to latest yunohost version which should fix restoration bug), logs here.

It's an error with package check. I fixed that after this PR.
(But I forgot to push my fix...)

> I tested an upgrade from a freshly installed instance (with current version), which was successful, but wordpress was still in 4.7 version... ?!

For this app, the upgrade process is managed by the app itself. But, with this upgrade, all the upgrade become fully automated.

**Comment by JimboJoe** [comment]
OK, this time package_check results are OK (?!), except the problem for restoration which is due to a bug in package_check itself.
So upgrade from last version was OK, and LGTM as well.

**Review by JimboJoe — APPROVED** [review]

**Comment by maniackcrudelis** [comment]
> except the problem for restoration which is due to a bug in package_check itself.

Are you sure your package check is up to date?
I just did a test and I do not have any errors.

**Comment by JimboJoe** [comment]
>>    except the problem for restoration which is due to a bug in package_check itself.
>
>Are you sure your package check is up to date?
>I just did a test and I do not have any errors.

Yes, it's up-to-date and I still got the error, see [here](https://framadrop.org/r/BA8ho_cnp2#0xAe2Cc9Epde6nV0u9JbJ9CLIYbuPRYiLaGZDj+2Ypk=).

**Comment by JimboJoe** [comment]
OK, after our chat, I forced a `git pull` on package_check and it appears it wasn't up-to-date.
Now package_check results are perfect 👍

**Review by frju365 — APPROVED** [review]
LGTM : OK

**Review by Rafi594 — APPROVED** [review]
I did a complete test everything work fine 

Thanks for this PR @maniackcrudelis !

File: scripts/restore [hunk 1]
```diff
@@ -37,7 +37,7 @@ db_name=$(ynh_app_setting_get $app db_name)
 # CHECK IF THE APP CAN BE RESTORED
 #=================================================
 
-sudo yunohost app checkurl "${domain}${path_url}" -a "$app" \
+yunohost app checkurl "${domain}${path_url}" -a "$app" \
```
Comment (JimboJoe) [inline scripts/restore]: One remaining fossil... :wink:

**Comment by maniackcrudelis** [comment]
Will be merged in 3 days on the testing branch.

**Pull request merged**